Team_Roster/immaculateGridCalculations/gridSolver.py

```diff
@@ -1,6 +1,110 @@
+from flask import jsonify, render_template, request
+from app import db
+import sqlalchemy as sa
+from app.models import People, Fielding, Batting, Team, Pitching, Season
+from sqlalchemy import func, and_
+
+
+def getAllTeams():
+    teams = db.session.query(Team.team_name).all() 
+    team_list = [team.team_name for team in teams]
+    return team_list
+
+
+def getPlayersByTeam(team_name):
+    print(f"Querying team: {team_name}")
+    players = (
+        db.session.query(People.playerID, Team.team_name)
+        .select_from(Team)
+        .join(Batting, (Batting.teamID == Team.teamID) & (Batting.yearID == Team.yearID))
+        .join(People, Batting.playerID == People.playerID)
+        .filter(Team.team_name == team_name)
+        .group_by(People.playerID, Team.team_name)
+    )
+    print(f"Players found for {team_name}: {players.count()}")
+    return players.all()
+
+
+def getPlayerWinsBySeason(numWins):
+    print(f"Querying players with >= {numWins} wins")
+
+    playerWins = (
+        db.session.query(
+            People.playerID
+        )
+        .join(Pitching, People.playerID == Pitching.playerID).join(Team, (Team.yearID == Pitching.yearID) & (Team.teamID == Pitching.teamID))
+        .filter(Pitching.p_W >= numWins)
+        .group_by(People.playerID)
+    )
+    print(f"Players found with >= {numWins} wins: {playerWins.count()}")
+
+    return playerWins.all()
+
+
 
 def solveGrid(questions):
```

Comment:
It is not the clearest which part of this function is doing what. Given that we will have so many people touching it, I'd appreciate it if you add comments explaining what everything is doing, and where and how everyone should add their queries.